### PR TITLE
Website Common Configuration

### PR DESCRIPTION
**Pull request by akestner**
**Issue, if available:** 
N/A

**Description of changes:**
This PR adds common configuration from the [Docsy Example site](https://github.com/google/docsy-example/blob/master/config.toml) that we'll be leveraging to build the Karpenter docs/landing page. 

It combines changes and resolves conflicts from [akestner/karpenter:website-landing](/akestner/karpenter/tree/website-landing) and [geoffcline/karpenter:preview-web](geoffcline/karpenter/tree/preview-web) so that each of these efforts (building the marketing landing page and technical documentation) can be completed independently and result in PRs that are primarily changes in content, not website config or scaffolding.

Once this is merged, @akestner and @geoffcline should rebase the existing branches, mentioned above, off of the new `HEAD` of [awslabs/karpenter:main](awslabs/karpenter/tree/main) to ensure the changes are incorporated and their branches' history is correct.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by akestner** [comment]
Apologies for the force push, had somehow not started this branch from `main`.

File: Makefile [hunk 1]
```diff
@@ -81,6 +81,9 @@ helm: ## Generate Helm Chart
 website: ## Generate Docs Website
 	cd website;npm install;hugo -d ../docs
 
+website: ## Generate Docs Website
+	cd website; npm install; git submodule update --init --recursive; hugo
+
```
Comment (geoffcline) [inline Makefile:86]: is this duplicating the same action?

File: Makefile [hunk 2]
```diff
@@ -81,6 +81,9 @@ helm: ## Generate Helm Chart
 website: ## Generate Docs Website
 	cd website;npm install;hugo -d ../docs
 
+website: ## Generate Docs Website
```
Comment (njtran) [inline Makefile:84]: Is there a difference between this target and the one above? Will doing `make website` hit both of these targets? Can we collapse this into one?

**Review by geoffcline — CHANGES_REQUESTED** [review]
is website action in makefile duplicated?

File: Makefile (hunk 2 above)
Comment (akestner) [reply Makefile:84]: Ahh apologies, bad merge resolution. I'll fix

File: Makefile (hunk 1 above)
Comment (akestner) [reply Makefile:86]: This is, a bad merge resolution artifact. Removing now

**Review by geoffcline — APPROVED** [review]

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -62,6 +62,16 @@ kubectl patch deployment karpenter -n karpenter --type='json' -p='[{"op": "repla
 open http://localhost:8080/metrics && kubectl port-forward service/karpenter-metrics -n karpenter 8080
 ```
 
+### Marketing and Docs Website
+The Karpenter marketing and docs site uses [Hugo](https://gohugo.io/) and [npm](https://www.npmjs.com/) to generate HTML markup from Markdown content and configuration files in `/website`. Hugo makes use of [`git submodules`](https://git-scm.com/book/en/v2/Git-Tools-Submodules) to bring in theme assets and configruation. The Karpenter website uses the [Docsy theme](https://www.docsy.dev/). To serve this website locally, ensure you have `hugo` and `npm` installed, have initialized and updated your local repository's submodules with `git submodule update --init --recursive`, and then use the follow commands to bring up a local webserver:
+```bash
+cd website
```
Comment (ellistarn) [inline DEVELOPER_GUIDE.md:68]: FYI you can do `make website`

File: Makefile [hunk 4]
```diff
@@ -79,7 +79,7 @@ helm: ## Generate Helm Chart
 	cd charts;helm lint karpenter;helm package karpenter;helm repo index .
 
 website: ## Generate Docs Website
-	cd website;npm install;hugo -d ../docs
+	cd website; npm install; git submodule update --init --recursive; hugo
```
Comment (ellistarn) [inline Makefile:82]: You need to make sure the destination is ../docs

Comment (geoffcline) [reply Makefile:82]: that is in the hugo config file now

**Pull request merged**